src/test/java/uk/gov/mca/beacons/service/accounts/AccountHolderControllerIntegrationTest.java

```diff
@@ -23,28 +26,64 @@
   void requestCreateAccountHolder_shouldRespondWithTheCreatedResource()
     throws Exception {
     String testAuthId = UUID.randomUUID().toString();
-    String newAccountHolderRequest = new String(
-      Files.readAllBytes(
-        Paths.get("src/test/resources/fixtures/createAccountHolderRequest.json")
-      )
+    String newAccountHolderRequest = readFile(
+      "src/test/resources/fixtures/createAccountHolderRequest.json"
     )
       .replace("replace-with-test-auth-id", testAuthId);
-    String expectedResponse = new String(
-      Files.readAllBytes(
-        Paths.get(
-          "src/test/resources/fixtures/createAccountHolderResponse.json"
-        )
-      )
+    String expectedResponse = readFile(
+      "src/test/resources/fixtures/createAccountHolderResponse.json"
     )
       .replace("replace-with-test-auth-id", testAuthId);
 
-    webTestClient
+    var response = webTestClient
+      .post()
+      .uri("/account-holder")
+      .body(BodyInserters.fromValue(newAccountHolderRequest))
+      .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+      .exchange()
+      .expectBody();
+
+    response.json(expectedResponse);
+    response.jsonPath("$.data.id").exists();
```

Comment:
`response.json(expectedResponse);` performs a [lenient](https://www.baeldung.com/jsonassert#1-using-the-lenient-mode) check of the response against the fixture.  The fixture doesn't contain an `id` field (because that's randomly generated by the production code), but the response does.  Lenient checking means this assertion only cares that the JSON paths that are in `expectedResponse` are there.

`response.jsonPath("$.data.id").exists();` mops up what's missing by checking the `id` field exists.  

There is no assertion to say that `id` is a UUID, for example.  I could spin something up with a Regex matcher but I think that'd be overkill for this case.

Thoughts?

Reply:
👍  Can change the `exists` `isNotEmpty` or similar to check the key exists and is not empty